include/mqtt/shared_ptr_array.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright Takatoshi Kondo 2019
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(MQTT_SHARED_PTR_ARRAY_HPP)
+#define MQTT_SHARED_PTR_ARRAY_HPP
+
+#if defined(_DOXYGEN_)
+
+/**
+ * @brief Type alias of shared_ptr char array.
+ * You can choose the target type.
+ * - If MQTT_STD_SHARED_PTR_ARRAY is defined, `std::shared_ptr<char []>` is used.
+ *   - std::shared_ptr<char []> is supported since C++17.
+ * - If MQTT_STD_SHARED_PTR_ARRAY is not defined (default), `boost::shared_ptr<char []>` is used.
+ *   - `boost::shared_ptr<char []>` can be used on C++14.
+ */
+using shared_ptr_array = std::shared_ptr<char []>;
+
+/**
+ * @brief shared_ptr_array creating function.
+ * You can choose the target type.
+ * - If MQTT_STD_SHARED_PTR_ARRAY is defined,
```

Comment:
The CMakeLists.txt file does not offer an option to control this behavior.

Was that something you planned to do?

Reply:
I forgot adding MQTT_STD_SHARED_PTR_ARRAY to CMakeLists.txt. I will add the following code to CMakeLists.txt.

```cmake
OPTION(MQTT_STD_SHARED_PTR_ARRAY "Use std::shared_ptr<char[]> from C++17 instead of boost::shared_ptr<char[]>" OFF)

...

IF (MQTT_STD_SHARED_PTR_ARRAY)
    SET(CMAKE_CXX_STANDARD 17)
    SET(CMAKE_CXX_STANDARD_REQUIRED ON)
    MESSAGE (STATUS "Using std::shared_ptr<char []> instead of boost::shared_ptr<char []>. Enables C++17!!!")
    SET (CMAKE_CXX_FLAGS "-DMQTT_STD_SHARED_PTR_ARRAY ${CMAKE_CXX_FLAGS}")
ELSE ()
    MESSAGE (STATUS "Using boost::shared_ptr<char []> instead of boost::shared_ptr<char []>")
ENDIF ()

```